Define _LARGEFILE_SOURCE and _FILE_OFFSET_BITS conditionally

In order to avoid ugly warnings at compile time, only define
_LARGEFILE_SOURCE and _FILE_OFFSET_BITS if they have not already been
defined through the build command line.

Avoids:

  In file included from redis.h:33:0,
                   from migrate.c:1:
  fmacros.h:45:0: warning: "_LARGEFILE_SOURCE" redefined
  <command-line>:0:0: note: this is the location of the previous definition

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: redis-2.6.11/src/fmacros.h
===================================================================
--- redis-2.6.11.orig/src/fmacros.h	2013-03-25 22:09:15.000000000 +0100
+++ redis-2.6.11/src/fmacros.h	2013-03-25 22:09:40.000000000 +0100
@@ -42,7 +42,12 @@
 #define _XOPEN_SOURCE
 #endif
 
+#ifndef _LARGEFILE_SOURCE
 #define _LARGEFILE_SOURCE
+#endif
+
+#ifndef _FILE_OFFSET_BITS
 #define _FILE_OFFSET_BITS 64
+#endif
 
 #endif
